metricbeat/mb/lightmodules.go

```diff
@@ -0,0 +1,288 @@
+		// Override default config on base module and metricset
+		base.name = m.Name
+		baseModule, err := m.baseModule(base.module)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create base module for light module")
+		}
+		base.module = baseModule
+
+		// Run module factory if registered, it will be called once per
+		// metricset, but it should be idempotent
+		moduleFactory := r.moduleFactory(m.Input.Module)
+		if moduleFactory != nil {
+			m, err := moduleFactory(*baseModule)
+			if err != nil {
+				return nil, err
+			}
+			base.module = m
+		}
+
+		// At this point host parser was already run, we need to run this again
+		// with the overriden defaults
+		if registration.HostParser != nil {
+			base.hostData, err = registration.HostParser(base.module, base.host)
+			if err != nil {
+				return nil, err
+			}
+			base.host = base.hostData.Host
+		}
+
+		return originalFactory(base)
+	}
+
+	return registration, nil
+}
+
+// baseModule does the configuration overrides in the base module configuration
+// taking into account the light metric set default configurations
+func (m *LightMetricSet) baseModule(from Module) (*BaseModule, error) {
+	baseModule := BaseModule{
+		name:   m.Module,
+		config: from.Config(),
+	}
+	var err error
+	// Set defaults
+	if baseModule.rawConfig, err = common.NewConfigFrom(m.Input.Defaults); err != nil {
+		return nil, errors.Wrap(err, "invalid input defaults")
+	}
+	// Copy values from user configuration
+	if err = from.UnpackConfig(baseModule.rawConfig); err != nil {
+		return nil, errors.Wrap(err, "failed to copy values from user configuration")
+	}
+	// Update module configuration
+	if err = baseModule.UnpackConfig(&baseModule.config); err != nil {
+		return nil, errors.Wrap(err, "failed to set module configuration")
+	}
+	return &baseModule, nil
+}
+
+func (r *LightModulesRegistry) readModule(moduleName string) (*LightModule, bool, error) {
+	modulePath, found := r.findModulePath(moduleName)
+	if !found {
+		return nil, false, nil
+	}
+
+	moduleConfig, err := r.readModuleConfig(modulePath)
+	if err != nil {
+		return nil, true, errors.Wrapf(err, "failed to load light module '%s' definition", moduleName)
+	}
+
+	metricSets := make(map[string]LightMetricSet)
+	for _, metricSet := range moduleConfig.MetricSets {
+		manifestPath := filepath.Join(filepath.Dir(modulePath), metricSet, manifestYML)
+
+		metricSetConfig, err := r.readMetricSetConfig(manifestPath)
+		if err != nil {
+			return nil, true, errors.Wrapf(err, "failed to load light metricset '%s/%s' definition", moduleName, metricSet)
+		}
+		metricSetConfig.Name = metricSet
+		metricSetConfig.Module = moduleName
+
+		metricSets[metricSet] = metricSetConfig
+	}
+
+	return &LightModule{Name: moduleName, MetricSets: metricSets}, true, nil
+}
+
+func (r *LightModulesRegistry) findModulePath(moduleName string) (string, bool) {
+	for _, dir := range r.paths {
+		p := filepath.Join(dir, moduleName, moduleYML)
```

Comment:
In Go is idiomatic to use single character arguments name and full name variables names in the body functions. At least give `p` a full name :smile: like `candidatePath`? I guess it's a string path it might or might not be correct because of later use of `Stat` method.